src/OrderFormAcceptanceTests.Steps/Steps/AssociatedServices.cs

```diff
@@ -238,6 +238,14 @@ public void GivenTheSupplierAddedToTheOrderHasAnAssociatedServiceWithADeclarativ
             order.Update(Test.ConnectionString);
         }
 
+        [Given(@"the supplier added to the order has an associated service with an on-demand flat price")]
+        public void GivenTheSupplierAddedToTheOrderHasAnAssociatedServiceWithAnOnDemandFlatPrice()
+        {
+            var order = (Order)Context["CreatedOrder"];
+            order.SupplierId = 100000;
+            order.SupplierName = "Really Kool Corporation";
+            order.Update(Test.ConnectionString);
+        }
```

Comment:
Do we have other steps that match this one?

Reply:
Not as much as I expected, after looking around. I've merged the two Associated Service ones into a single step though